daemon/algod/api/server/v2/handlers.go

```diff
@@ -944,10 +944,23 @@ type PreEncodedSimulateTxnGroupResult struct {
 
 // PreEncodedSimulateResponse mirrors model.SimulateResponse
 type PreEncodedSimulateResponse struct {
-	Version      uint64                             `codec:"version"`
-	LastRound    uint64                             `codec:"last-round"`
-	TxnGroups    []PreEncodedSimulateTxnGroupResult `codec:"txn-groups"`
-	WouldSucceed bool                               `codec:"would-succeed"`
+	Version       uint64                             `codec:"version"`
+	LastRound     uint64                             `codec:"last-round"`
+	TxnGroups     []PreEncodedSimulateTxnGroupResult `codec:"txn-groups"`
+	WouldSucceed  bool                               `codec:"would-succeed"`
+	LiftLogLimits *bool                              `codec:"lift-log-limits,omitempty"`
+	EvalChanges   *PreEncodedSimulateEvalChanges     `codec:"eval-changes,omitempty"`
+}
+
+// PreEncodedSimulateEvalChanges mirrors `eval-changes` field in model.SimulateResponse
+type PreEncodedSimulateEvalChanges struct {
```

Comment:
I think you can/should directly use `model.SimulateEvalChanges` here.

The reason we must create a new `PreEncodedSimulateResponse` is because it needs `PreEncodedSimulateTxnGroupResult` which needs `PreEncodedSimulateTxnResult` which needs `PreEncodedTxInfo` which needs a `transactions.SignedTxn` instead of the generated `map[string]interface{}`. Similarly, `PreEncodedSimulateRequest` needs `PreEncodedSimulateRequestTransactionGroup` which needs `transactions.SignedTxn` instead of a more generic object.

Needing `transactions.SignedTxn` for our code but not wanting to commit to it in the REST API is the root cause of all of the struct copying shenanigans.

Reply:
fair, let me take these additional struct down.

Reply:
see [refactor Simulate interface, remove unnecessary codec struct, remove …](https://github.com/algorand/go-algorand/pull/5247/commits/61d8b7e19f08848a812d01c8933950740c97b60d)